### PR TITLE
TS-2482: Should use target_addr instead of server_addr when build socks request.

### PR DESCRIPTION
**Pull request by oknet**

### CODE REVIEW

**Comment by mingzym** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/633/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/529/ for details.

File: iocore/net/Socks.cc [hunk 1]
```diff
@@ -136,7 +136,7 @@ void
 SocksEntry::free()
 {
   MUTEX_TRY_LOCK(lock, action_.mutex, this_ethread());
-  if (lock.is_locked()) {
+  if (!lock.is_locked()) {
     // Socks continuation share the user's lock
     // so acquiring a lock shouldn't fail
     ink_assert(0);
```
Comment (zwoop) [inline iocore/net/Socks.cc]: This reads a little strange, and has contradicting semantics in debug build (where it'll abort) vs release builds (where the call to SocksEntry::free() silently fails).

Would it make more sense to change this to an ink_release_assert()? E.g.

    ink_release_assert(lock.is_locked);

? If not, then the invariant is not strong enough, and likely should be changed to not be an assert at all, right?

**Comment by zwoop** [comment]
The changes looks good to me.

**Comment by oknet** [comment]
@zwoop ink_release_assert added.

**Comment by jpeach** [comment]
@oknet FWIW if you need to assert unconditionally, you can use ``ink_abort``, which just aborts with a message.

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/634/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/530/ for details.

**Comment by oknet** [comment]
@jpeach It is conditional assert on lock check. Do u means keep the if statement and ink_abort inside it ?

**Comment by jpeach** [comment]
@oknet the code is fine, it was just a FYI :)

**Comment by zwoop** [comment]
:+1:

**Pull request merged**